### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by MinhyukPark**
This PR adds a login page and fixes a lot of issues such as swapping row-column labels and fixing formula bugs :(

---
This change is [Reviewable](https://reviewable.io/reviews/minhyukpark/dataspreadfrontend/15)

### CODE REVIEW

**Comment by MinhyukPark** [comment]
___
*[src/components/app/index.js, line 764 at r1](https://reviewable.io/reviews/minhyukpark/dataspreadfrontend/15#-M18HtCICzLs3ZI3iKm7:-M18HtCICzLs3ZI3iKm8:b3ahhz9) ([raw file](https://github.com/minhyukpark/dataspreadfrontend/blob/a41fef898084b5f47cf79ae94c1e740668ff1bfe/src/components/app/index.js#L764)):*
> ```JavaScript
> 
> 
>     handleRowDelete(event) {
> ```

ah we should change this to the new format huh

File: src/components/adddbpage/index.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+// relative imports
+import "~/components/adddbpage/adddbpage.css";
+import { isDebug } from "~/utils/spreadsheet_utils/";
+
+class AddDBPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
```
Comment (laurenho025) [inline src/components/adddbpage/index.js]: Did we decide if we are binding functions manually or using arrow functions? The style guide mentions to use arrow functions.

File: src/components/adddbpage/index.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    handleChange(e, {name, value}) {
+        this.setState({
+            [name]: value,
+        });
+    }
+
+    handleSubmit() {
+        if (isDebug()) {
+            console.log("here is handleSubmit, isDebug")
```
Comment (laurenho025) [inline src/components/adddbpage/index.js]: Small thing but console.log is still here

File: src/components/adddbpage/index.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                sqltype: "mysql",
+                sqlhost: "localhost",
+                sqlport: "3306",
+                sqluser: "dataspread-user-1",
+                sqlpassword: "dataspread-pass-1",
+                sqldatabase: "dataspread_local",
+                sqltable: "demo",
+            });
+        } else {
+            console.log("here is handleSubmit, else")
```
Comment (laurenho025) [inline src/components/adddbpage/index.js]: console.log again

File: src/components/app/index.js [hunk 4]
```diff
@@ -388,29 +390,24 @@ class App extends React.Component {
...
+              for (let i = 0; i < rowLength; i++) {
+                for (let j = 0; j < colLength; j++) {
+                  if (currentData !== undefined && currentData[i] !== undefined) {
+                    currentString += (`${currentData[i][j]},`);
+                  }
+                }
               }
-            }
+              console.log(`toBeEvaluated[currentKey] : ${toBeEvaluated[currentKey]}`)
+              console.log(`currentString : ${currentString}`)
```
Comment (laurenho025) [inline src/components/app/index.js]: console.logs again

File: src/components/formulabar/index.js [hunk 5]
```diff
@@ -28,6 +28,7 @@ class FormulaBar extends React.Component {
     const { formulaRowIndex, formulaColumnIndex } = this.props;
     console.log('exiting formula bar');
     if (e.target !== undefined) {
+      console.log(`selected %v, %v`, formulaRowIndex, formulaColumnIndex);
```
Comment (laurenho025) [inline src/components/formulabar/index.js]: console.log here

**Pull request merged**